meson.build

```diff
@@ -0,0 +1,237 @@
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+endif
+if iconv_dep.found() or cc.has_function('iconv_open')
+    conf.set('CONFIG_ICONV', 1)
+endif
+
+freetype_dep = dependency('freetype2',version: '>= 2.10.92',
+                          fallback: ['freetype2', 'freetype_dep'],
+                          default_options: ['default_library=' + get_option('default_library'), 'harfbuzz=disabled']) # workaround for https://github.com/mesonbuild/meson/issues/8047
+if freetype_dep.found()
+    deps += freetype_dep
+    conf.set('CONFIG_FREETYPE', 1)
+endif
+
+if host_machine.system() == 'windows' and get_option('default_library') == 'static'
+    add_project_arguments('-DFRIBIDI_LIB_STATIC', language: 'c')
+endif
+fribidi_dep = dependency('fribidi', version: '>= 0.19.1',
+                         fallback: ['fribidi', 'libfribidi_dep'],
+                         default_options: ['docs=false', 'tests=false'])
+if fribidi_dep.found()
+    deps += fribidi_dep
+    conf.set('CONFIG_FRIBIDI', 1)
+endif
+
+harfbuzz_options = ['tests=disabled', 'cairo=disabled', 'gobject=disabled']
+if host_machine.system() == 'windows' or host_machine.system() == 'darwin'
+    harfbuzz_options += ['glib=disabled', 'freetype=disabled']
+endif
+harfbuzz_dep = dependency('harfbuzz', version: '>= 1.2.3',
+                          fallback: ['harfbuzz', 'libharfbuzz_dep'],
+                          default_options: harfbuzz_options)
+if harfbuzz_dep.found()
+    deps += harfbuzz_dep
+    conf.set('CONFIG_HARFBUZZ', 1)
+endif
+
+fontconfig_dep = dependency('fontconfig', version: '>= 2.10.92', required: get_option('fontconfig'))
+fontconfig = fontconfig_dep.found()
+if fontconfig
+    deps += fontconfig_dep
+    conf.set('CONFIG_FONTCONFIG', 1)
+endif
+
+if get_option('test')
+    deps += dependency('libpng', version: '>= 1.2.0')
+    conf.set('CONFIG_LIBPNG', 1)
+endif
+
+# DirectWrite
+directwrite = false
+if host_machine.system() == 'windows' and not get_option('directwrite').disabled()
+    if cc.has_header('windows.h')
+        deps += cc.find_library('dwrite', required: true)
+        conf.set('CONFIG_DIRECTWRITE', 1)
+        directwrite = true
+    elif get_option('directwrite').enabled()
+        error('DirectWrite was enabled explicitly, but required header is missing.')
+    endif
+endif
+
+# CoreText
+coretext = false
+if host_machine.system() == 'darwin' and not get_option('coretext').disabled()
+    coretext_dep = dependency('appleframeworks', modules: ['CoreText', 'CoreFoundation'], required: false)
+    if cc.has_type('CTRunRef', prefix: '#include <CoreText/CoreText.h>', dependencies: coretext_dep)
+        deps += coretext_dep
+        conf.set('CONFIG_CORETEXT', 1)
+        coretext = true
+    elif get_option('coretext').enabled()
+        error('CoreText was enabled explicitly, but required headers or frameworks are missing.')
+    endif
+endif
```

Comment:
Iinm this doesn't deal with the old include style with `ApplicationServices`.

Reply:
I want to say this was intentional for some reason, but I'll have to read back up. Maybe Meson didn't behave correctly with it? Though I have no idea how to even test it, since I don't own a sufficiently old Apple device or even know someone with a sufficiently old Apple device.

Also, hasn't the Coretext option been around for 15 years or something at this point? Is there a reason you don't just mandate that even in the normal build?

Reply:
That’s fair. But there’s little reason to suspect that it would suddenly stop working as long as we don’t start using APIs that simply don’t exist on that macOS version (which is wel-documented).

In specific practical matters, I used to run Snow Leopard (which the “old include style” was meant to support) myself for a long time, so we know this used to work, though I finally upgraded several years ago. And technically, if significant testing were to ever be needed, I believe I can still make a VM of it fairly easily, though I’m currently also migrating away from Macs entirely; of course, you’re right in that I’d rather avoid it.

Of course, rejecting old macOS _entirely_ wouldn’t be our cup of tea as long as you can still build libass and our dependencies on it. Even if not Core Text, there’s always Fontconfig. I mean, we don’t generally do the thing where we have a few selected platforms that we “support” and anything else does not concern us. Quite the opposite; we strive to run anywhere. We do have some hardware/compiler requirements such as two’s complement 32+bit `int` (…if we even do still assume that anywhere), but we try to keep that to a minimum. Of course, again, this doesn’t mean we have to go out of our way to support _an extra feature_ on an ancient macOS, but in this case it’s so easy that I see no harm.

Reply:
I'm adding an attempt at the `ApplicationServices` check, but like I mentioned, I have no way to check if it actually works. You might want to run it in the VM to make sure it isn't just broken.